crates/circuit/src/dag_circuit.rs

```diff
@@ -234,9 +234,9 @@ pub struct DAGCircuit {
     cregs: Py<PyDict>,
 
     /// The cache used to intern instruction qargs.
-    qargs_interner: Interner<[Qubit]>,
+    pub qargs_interner: Interner<[Qubit]>,
     /// The cache used to intern instruction cargs.
-    cargs_interner: Interner<[Clbit]>,
+    pub cargs_interner: Interner<[Clbit]>,
```

Comment:
I'm assuming that you're keeping these as public because you're adding instructions manually to the `DAGCircuit`. Since #13143 merged, you can add operations to the circuit without needing to manually intern your qargs/cargs.
```suggestion
    /// The cache used to intern instruction qargs.
    qargs_interner: Interner<[Qubit]>,
    /// The cache used to intern instruction cargs.
    cargs_interner: Interner<[Clbit]>,
```
If there's another reason you need these to remain public comment here with your thoughts.

Reply:
I think this is the main reason I found: https://github.com/Qiskit/qiskit/pull/13141#discussion_r1782434049

Reply:
I'm okay with leaving these as public for the time being, however. I'd like to hear if @kevinhartman agrees.